projects/concurrency/cache_test.go

```diff
@@ -0,0 +1,101 @@
+package concurrency
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func TestCache(t *testing.T) {
+	t.Run("initializes a cache", func(t *testing.T) {
+		c := NewCache[string, any](3)
+		require.Equal(t, c.entryLimit, 3)
+		require.NotNil(t, c.cache)
+	})
+
+	t.Run("add item to a cache", func(t *testing.T) {
+		c := NewCache[string, any](3)
+		ok := c.Put("name", "John")
+		require.False(t, ok)
+		val, ok := c.cache["name"]
+		require.Equal(t, "John", val.value)
+		require.True(t, ok)
+	})
+
+	t.Run("get item from a cache", func(t *testing.T) {
+		c := NewCache[string, any](3)
+
+		val, ok := c.Get("name")
+		require.False(t, ok)
+		require.Nil(t, val)
+
+		c.Put("name", "John")
+
+		val, ok = c.Get("name")
+		require.True(t, ok)
+		require.NotNil(t, val)
+		require.Equal(t, "John", *val)
+	})
+
+	t.Run("does not exceed entryLimit", func(t *testing.T) {
+		c := NewCache[string, any](3)
+		input := map[string]any{
+			"name":   "John",
+			"age":    30,
+			"hobby":  "swimming",
+			"friend": "Amy",
+		}
+		for key, value := range input {
+			c.Put(key, cacheEntry[any]{value: value, timeStamp: time.Now()})
+		}
+		require.Equal(t, 3, len(c.cache))
```

Comment:
I would probably insert things in a known order, and then assert which of them can still be `Get`'d and which can't - that tests not just that we _enforce_ the entry limit, but also that we evict the correct values when we hit it.